src/simpla-collection-editor.html

```diff
@@ -331,42 +336,15 @@
        * @return {undefined}
        */
       _removeItem(e) {
-        let itemData = Polymer.dom(e).rootTarget
-              .closest(`[${ITEM_DATA_ATTR}]`)
-              .getAttribute(ITEM_DATA_ATTR),
-            item = JSON.parse(itemData);
-
-        this.fire('collection-item-edit', {
-          action: 'remove',
-          args: [ item.id ]
-        });
+        this.splice('items', e.model.index, 1);
```

Comment:
@seaneking didn't know if you knew this one, but if capture an event from inside a dom-repeat it'll have a `model` property which will have a bunch of handy properties, including the index of the item that was clicked. Also the item itself, so I could have also done: `let itemToRemove = e.model.item`